ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp

```diff
@@ -956,10 +956,10 @@ void Matmul::validate(
         (input_tensor_a.get_layout() == Layout::TILE && input_tensor_b.get_layout() == Layout::TILE),
         "Inputs to matmul must be tilized");
     TT_FATAL(
-        a_shape[-1] == b_shape[-2],
+        a_shape.with_tile_padding()[-1] == b_shape.with_tile_padding()[-2],
```

Comment:
You cannot change the assertion.

The shapes need to match, not the padded shapes.

Reply:
This was causing an issue [here](https://github.com/tenstorrent/tt-metal/issues/12141).

The problem was that a_shape contained the original shape, but b_shape contained the padded shape. 

The same issue had previously occurred with the bias shape as well, which was fixed [here](https://github.com/tenstorrent/tt-metal/blob/3216f4bbc151c6cd20bffecba31c87450753118b/ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp#L1001).

Reply:
Then you need to fix this at the source of a and b.

a vs b shape is different than b vs bias, which I still need to double check.

Reply:
Looked into the other check. The check isn't right for bias either. Created https://github.com/tenstorrent/tt-metal/issues/13058 for me to fix that.